charmdir.go

```diff
@@ -454,19 +453,69 @@ func (m NopLogger) Infof(message string, args ...interface{}) {
 
 }
 
+type vcsCMD struct {
+	vcsType       string
+	args          []string
+	usesTypeCheck func() bool
+	errHandler    func(err error, vcsType string, path string)
+	charm         *CharmDir
+}
+
+// git is the most used vcs
+// we want to know whether the underlying system is using git
+func usesGit() bool {
+	args := []string{"rev-parse", "--is-inside-work-tree"}
+	cmd := exec.Command("git", args...)
+	if _, err := cmd.Output(); err == nil {
+		return true
+	}
+	return false
+}
+
+func handleErrGit(err error, vcsType string, charmPath string) {
+	logger.Debugf("%q version string generation failed : %v\nThis means that the charm version won't show in juju status. Charm path %q", vcsType, err, charmPath)
```

Comment:
I'm unsure what the need for `vcsType` here is, as you know you're handling `git`.